### DISCUSSION TITLE
Metrics with Version

### DISCUSSION THREAD

**Discussion by genofire** [fluxcd/flux2#3769] [Proposals] [answered]
It would be nice, to have in the metrics also a n installed version.

eg.

`gotk_condition{type="HelmRelease",version="1.3.2",name="kube-prom",namespace="mon"} 1`

**Comment by stefanprodan** [comment]
You could use https://github.com/sstarcher/helm-exporter to get Helm specific metrics.

The GitOps Toolkit metrics are generic, they are exposed by all Flux controllers for every custom resource. The `version` attribute is specific to Helm only, it does not exist for GitRepositories, ImagePolicies, Kustomizations, etc. Adding labels such as `version`, `digest`, `checksum`, which change in time, implies high cardinality.  Given this, I don’t think we can add `version` to `gotk_condition`.

Maybe we could introduce custom metrics specific to a controller? e.g. 
`flux_helm_release{version="1.3.2",name="cube-prom",namespace="mon"}`.

Having custom metrics can significantly increase the Prometheus storage size of Flux data. We should probably make this opt-in via a feature flag.

**Comment by kingdonb** [comment, marked as answer]
Please refer to:

* https://github.com/fluxcd/flux2/issues/4128

This has been addressed, now you can create any metrics that you wish if the information is in the CRD spec or status, you can create a metric to report on it. That includes version, there is an example in the docs to support it.

See also:

https://fluxcd.io/flux/monitoring/custom-metrics/
